### PR TITLE
Add scala tensorflow load/save examples 

### PR DESCRIPTION
**Pull request by yiheng**
## What changes were proposed in this pull request?
* Add scala examples to show how to load/save tensorflow model
* Refactor error message when cannot load tensorflow model
* fix some bugs when convert tensorflow model to bigdl
* put more scripts to bin folder when make a dist

## How was this patch tested?
Manual test the example and pr validation

## Related links or issues (optional)
fix https://github.com/intel-analytics/BigDL/issues/1034

### CODE REVIEW

File: scripts/download.sh [hunk 1]
```diff
@@ -1,4 +1,27 @@
-#!/bin/sh                               
+#!/bin/sh
```
Comment (yiheng) [inline scripts/download.sh]: just add license header

File: scripts/dump_tf_graph.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#
```
Comment (yiheng) [inline scripts/dump_tf_graph.py]: dump tensorflow model summary and show it in tensorboard

File: spark/dist/assembly/dist.xml [hunk 3]
```diff
@@ -19,6 +19,10 @@
       <includes>
           <include>classes.lst</include>
           <include>img_class.lst</include>
+          <include>download.sh</include>
```
Comment (yiheng) [inline spark/dist/assembly/dist.xml]: Issue #1034

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/example/tensorflow/.gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+model/
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/example/tensorflow/.gitignore]: ignore some test files

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowToBigDL.scala [hunk 5]
```diff
@@ -396,6 +396,71 @@ object Conv2D extends TensorflowToBigDL{
...
+    require(attributes.get("strides").getList.getI(0).toInt == 1, s"not support strides on batch")
+    require(attributes.get("data_format").getS.toString(Charset.defaultCharset()) == "NCHW",
+      "NCHW should be used for this sub-graph")
+
+    require(attributes.get("strides").getList.getI(2).toInt == 1, s"not support strides on depth")
+    val (strideH, strideW) = (attributes.get("strides").getList.getI(2).toInt,
+      attributes.get("strides").getList.getI(3).toInt)
+
+    val biasNode = tfGraph.source.prevNodes(1).prevNodes(0).prevNodes.head.element
+    val (bias, gradBias) = getOrSetTensor(biasNode, context, byteOrder)(t => t)
```
Comment (yangw1234) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowToBigDL.scala]: It seems the `Reshape` operation is not reflected on the BigDL model. Should we impose some requirement on the shape of this biasTensor?

**Comment by yangw1234** [comment]
LGTM except one minor concern on the new convolution mapping

**Comment by yiheng** [comment]
PR validation pass.

**Comment by yiheng** [comment]
As discussed, merge it.

**Pull request merged**